interfaces/Portalicious/src/app/auth.guard.ts

```diff
@@ -18,3 +19,30 @@ export const authGuard: CanActivateFn = (_, state) => {
     queryParamsHandling: 'merge',
   });
 };
+
+export const projectPermissionsGuard: (
+  permission: PermissionEnum,
+) => CanActivateFn = (permission: PermissionEnum) => {
+  return function projectPermissionsCanActivateFn(
+    route: ActivatedRouteSnapshot,
+  ) {
+    const authService = inject(AuthService);
+
+    const projectId = route.params.projectId as number;
+
+    if (isNaN(projectId)) {
+      throw new Error('projectId is not a number');
+    }
+
+    if (authService.hasPermission(projectId, permission)) {
+      return true;
+    }
+
+    const router = inject(Router);
+    return router.navigate(['/', AppRoutes.project, projectId], {
+      queryParams: {
+        permissionDenied: true,
```

Comment:
Is this used somewhere already? Or is there a default-feature of Angular for this?

Reply:
No. It's something that I forgot to implement 🙈 I will add a TODO message for now and we can decide whether to look at it now or revisit later.

Reply:
Ah, yes..  The whole flow of "your session/token/cookie has expired and you need to re-login" needs to be looked at, also with SSO in mind, would deserve its own backlog-item, I think.

Reply:
Oh, wait... there is also this error-pattern:

https://github.com/global-121/121-platform/pull/5381/files?file-filters%5B%5D=.html&file-filters%5B%5D=.ts&show-viewed-files=true#diff-87ffb49babf9b145819c6fdb333f5e849b84c120e52fab27ae5b7b7e430e2a89

Where we transfer 1 (out of 2/multiple)-error-coded between pages/reloads via local-storage...

Reply:
I'm resolving this assuming that we will address this weird pattern when we address the TODO